awx/main/tasks/jobs.py

```diff
@@ -440,19 +440,15 @@ def final_run_hook(self, instance, status, private_data_dir):
         """
         instance.log_lifecycle("finalize_run")
         artifact_dir = os.path.join(private_data_dir, 'artifacts', str(self.instance.id))
-        collections_info = os.path.join(artifact_dir, 'collections.json')
-        ansible_version_file = os.path.join(artifact_dir, 'ansible_version.txt')
-
-        if os.path.exists(collections_info):
-            with open(collections_info) as ee_json_info:
-                ee_collections_info = json.loads(ee_json_info.read())
-                instance.installed_collections = ee_collections_info
-                instance.save(update_fields=['installed_collections'])
-        if os.path.exists(ansible_version_file):
-            with open(ansible_version_file) as ee_ansible_info:
-                ansible_version_info = ee_ansible_info.readline()
-                instance.ansible_version = ansible_version_info
-                instance.save(update_fields=['ansible_version'])
+        data_file_path = os.path.join(artifact_dir, 'ansible_data.json')
+
+        if os.path.exists(data_file_path):
+            with open(data_file_path) as f:
+                collected_data = json.loads(f.read())
+
+            instance.installed_collections = collected_data['installed_collections']
```

Comment:
This could throw a `ValueError` if the plugin returns invalid or out-of-spec json.